dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/AttributeValidationHook.java

```diff
@@ -160,4 +151,29 @@ protected void validateAttributeUniqueness( ValidationErrorReporter errorReporte
         }
     }
 
+    private TrackerErrorCode getTrackerErrorCode( ErrorMessage errorMesage )
+    {
+        switch ( errorMesage.getErrorCode() )
+        {
+        case E2027:
+            return TrackerErrorCode.E1084;
+        case E2029:
+            return TrackerErrorCode.E1125;
+        case E2030:
+            return TrackerErrorCode.E1085;
+        case E2040:
+            return TrackerErrorCode.E1101;
+        case E2041:
+            return TrackerErrorCode.E1105;
+        case E2042:
+            return TrackerErrorCode.E1106;
+        case E2043:
+            return TrackerErrorCode.E1304;
+        case E2026:
+            return TrackerErrorCode.E1009;
+        default:
+            return null;
+        }
+    }
+
```

Comment:
I have some concerns about this part.
`ValueTypeValidationUtils` is validating the value type and returning some error codes if something is wrong.
These error codes only really make sense for the old tracker importer, to be used in the new tracker importer we need to convert them.
First conceptual problem is that we are coupling the new importer to the old one throw these codes.
Second problem is that if one day we remove the old importer, we will have a validator that returns error codes that are not really used by any code in the system, but they still need to be converted to be used in the NTI.

If we plan at any point in the future to remove the old importer we can either duplicate the code for this validator, so when the old importer is gone one of the duplicates is gone or we can abstract the validator to not really know about the codes and then give the old and new importer the responsibility to use the right code based on the error reported by the validator

Reply:
Very good point 👍🏻 

If ValueTypeValidationUtils is only used by new and old tracker than duplicating it with error codes appropriate to each package/domain seems easy and clean.